pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -34,6 +34,10 @@ in
 , enableStaticLibraries ? true
 , enableHsc2hsViaAsm ? hostPlatform.isWindows && stdenv.lib.versionAtLeast ghc.version "8.4"
 , extraLibraries ? [], librarySystemDepends ? [], executableSystemDepends ? []
+# On macOS, statically linking against system frameworks is not supported;
+# see https://developer.apple.com/library/content/qa/qa1118/_index.html
+# They must be propagated to the environment of any executable linking with the library
+, libraryDarwinFrameworkDepends ? [], executableDarwinFrameworkDepends ? []
```

Comment:
Cabal refers to these dependencies simply as `frameworks` [in its `BuildInfo` type](http://hackage.haskell.org/package/Cabal-2.2.0.1/docs/Distribution-PackageDescription.html#t:BuildInfo), therefore I think we should adopt that nomenclature and simply call them `framework` here, too, rather than `darwinFramework`.